sdk/cosmos/azure-cosmos/azure/cosmos/_change_feed/change_feed_state.py

```diff
@@ -0,0 +1,408 @@
+            if version is None:
+                raise ValueError("Invalid base64 encoded continuation string [Missing version]")
+
+            if version == "V2":
+                return ChangeFeedStateV2.from_continuation(container_link, container_rid, continuation_json)
+
+            raise ValueError("Invalid base64 encoded continuation string [Invalid version]")
+
+        # when there is no continuation token, by default construct ChangeFeedStateV2
+        return ChangeFeedStateV2.from_initial_state(container_link, container_rid, change_feed_state_context)
+
+class ChangeFeedStateV1(ChangeFeedState):
+    """Change feed state v1 implementation.
+     This is used when partition key range id is used or the continuation is just simple _etag
+    """
+
+    def __init__(
+            self,
+            container_link: str,
+            container_rid: str,
+            change_feed_start_from: ChangeFeedStartFromInternal,
+            partition_key_range_id: Optional[str] = None,
+            partition_key: Optional[Union[str, int, float, bool, List[Union[str, int, float, bool]], _Empty, _Undefined]] = None, # pylint: disable=line-too-long
+            continuation: Optional[str] = None):
+
+        self._container_link = container_link
+        self._container_rid = container_rid
+        self._change_feed_start_from = change_feed_start_from
+        self._partition_key_range_id = partition_key_range_id
+        self._partition_key = partition_key
+        self._continuation = continuation
+
+    @property
+    def container_rid(self):
+        return self._container_rid
+
+    @classmethod
+    def from_json(
+            cls,
+            container_link: str,
+            container_rid: str,
+            change_feed_state_context: Dict[str, Any]) -> 'ChangeFeedStateV1':
+        return cls(
+            container_link,
+            container_rid,
+            ChangeFeedStartFromInternal.from_start_time(change_feed_state_context.get("startTime")),
+            change_feed_state_context.get("partitionKeyRangeId"),
+            change_feed_state_context.get("partitionKey"),
+            change_feed_state_context.get("continuationPkRangeId")
+        )
+
+    def populate_request_headers(
+            self,
+            routing_provider: SmartRoutingMapProvider,
+            request_headers: Dict[str, Any]) -> None:
+        request_headers[http_constants.HttpHeaders.AIM] = http_constants.HttpHeaders.IncrementalFeedHeaderValue
+
+        # When a merge happens, the child partition will contain documents ordered by LSN but the _ts/creation time
+        # of the documents may not be sequential. So when reading the changeFeed by LSN,
+        # it is possible to encounter documents with lower _ts.
+        # In order to guarantee we always get the documents after customer's point start time,
+        # we will need to always pass the start time in the header.
+        self._change_feed_start_from.populate_request_headers(request_headers)
```

Comment:
Will this be futile anyway since `ChangeFeedStateV1` is not merge-proof?

Reply:
good point - maybe can remove it from V1 version